mucommander-registry/src/main/java/com/mucommander/commons/file/protocol/registry/RegistryImage.java

```diff
@@ -0,0 +1,75 @@
+package com.mucommander.commons.file.protocol.registry;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.Objects;
+
+import com.mucommander.commons.file.FileFactory;
+import com.mucommander.commons.file.archive.tar.TarArchiveFile;
+import com.mucommander.commons.file.util.PathUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.mucommander.commons.file.AbstractFile;
+import com.mucommander.commons.file.FileURL;
+
+public class RegistryImage extends AbstractRegistryImage {
```

Comment:
correct me if I'm wrong - but that kind of mixes the protocol and the archive together, right?
wouldn't it make more sense to separate them out? meaning, to have this protocol-plugin listing, downloading and uploading images from the repositories and have an archive plugin for manipulating the images?

Reply:
no need, as we don't actually listing images, but rather downloading a specific image and display its layers.